go-controller/pkg/ovn/egressip.go

```diff
@@ -1609,18 +1259,30 @@ func (oc *DefaultNetworkController) generateCacheForEgressIP() (map[string]egres
 	for _, egressIP := range egressIPs {
 		egressIPCache[egressIP.Name] = egressIPCacheEntry{
 			egressPods:       make(map[string]sets.Set[string]),
-			gatewayRouterIPs: sets.New[string](),
+			gatewayRouterIPs: sets.New[string](), // can be transit switchIPs for interconnect multizone setup
 			egressIPs:        map[string]string{},
 		}
+		isEgressNodeLocal := false
 		for _, status := range egressIP.Status.Items {
+			var nextHopIP string
 			isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
-			gatewayRouterIP, err := oc.eIPC.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
-			if err != nil {
-				klog.Errorf("Unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %v", status.Node, isEgressIPv6, err)
-				continue
+			_, ok := oc.localZoneNodes.Load(status.Node)
+			if !config.OVNKubernetesFeature.EnableInterconnect || ok {
+				gatewayRouterIP, err := oc.eIPC.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
+				if err != nil {
+					klog.Errorf("Unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %v", status.Node, isEgressIPv6, err)
+					continue
+				}
+				nextHopIP = gatewayRouterIP.String()
+			} else {
+				nextHopIP, err = oc.eIPC.getTransitSwitchIP(status.Node)
+				if err != nil {
+					return nil, fmt.Errorf("unable to fetch transit switch IP for node %s: %v", status.Node, err)
```

Comment:
Any reason why failing to find join is a `continue`, while transit is a `return`?

Reply:
yeah good point. its because earlier we were continuing and then I thought should we really continue here or return ? I am torn between strongly faling versus silently ignoring.. I guess continue is the right approach because there will be a brief period of time when probably node's annotations are not yet added? I'll stick with what we were doing before unless someone feels otherwise.

Reply:
ack. my suggestion would be to put a comment here explaining that reasoning. it would be good to be consistent in how to handle this.

Reply:
yes I changed this into using continue as well so that its like the other parts of the code...